src/main/java/com/onarandombox/MultiverseCore/listeners/MVPlayerListener.java

```diff
@@ -30,26 +39,29 @@
 import org.bukkit.event.player.PlayerPortalEvent;
 import org.bukkit.event.player.PlayerRespawnEvent;
 import org.bukkit.event.player.PlayerTeleportEvent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 /**
  * Multiverse's {@link Listener} for players.
  */
 public class MVPlayerListener implements Listener {
     private final MultiverseCore plugin;
     private final MVWorldManager worldManager;
-    private final PermissionTools pt;
+    private final PlayerActionChecker actionChecker;
 
-    private final Map<String, String> playerWorld = new ConcurrentHashMap<String, String>();
+    private final Map<String, String> playerWorld = new ConcurrentHashMap<>();
 
     public MVPlayerListener(MultiverseCore plugin) {
         this.plugin = plugin;
         worldManager = plugin.getMVWorldManager();
-        pt = new PermissionTools(plugin);
+        actionChecker = plugin.getPlayerActionChecker();
     }
 
     /**
      * @return the playerWorld-map
      */
+    @Deprecated
     public Map<String, String> getPlayerWorld() {
         return playerWorld;
     }
```

Comment:
What was this for and what is replacing it?

Reply:
I dont get the need for tracking what world the player is in, when you can literally do Player#getWorld(). I believe its some legacy thing that should be removed.

Reply:
That's fair I guess. It is a public method though, so it might be in use in other plugins. I guess if it is a problem, then this method should have been named better because if it's what it says on the tin then they can just use the method you said.

Reply:
yea, will address this issue with a seperate PR. Either ways the API will be breaking, so it will either be refactor out of playerListener class or removed if I still cant see the purpose of it.